src/bundle.c

```diff
@@ -1641,6 +1641,147 @@ gboolean check_bundle_payload(RaucBundle *bundle, GError **error)
 	return res;
 }
 
+gboolean replace_signature(RaucBundle *bundle, const gchar *insig, const gchar *outpath, CheckBundleParams params, GError **error)
+{
+	g_autoptr(RaucManifest) manifest = NULL;
+	g_autoptr(RaucBundle) outbundle = NULL;
+	g_autoptr(GFile) bundleoutfile = NULL;
+	GFileIOStream* bundlestream = NULL;
+	GOutputStream* bundleoutstream = NULL;
+	g_autoptr(GBytes) sig = NULL;
+	gchar* keyringpath = NULL;
+	gchar* keyringdirectory = NULL;
+	GError *ierror = NULL;
+	gboolean res = FALSE;
+	gsize sigsize;
+
+	g_return_val_if_fail(bundle != NULL, FALSE);
+	g_return_val_if_fail(outpath != NULL, FALSE);
+	g_return_val_if_fail(insig != NULL, FALSE);
+	g_return_val_if_fail(error == NULL || *error == NULL, FALSE);
+
+	if (!(params & CHECK_BUNDLE_NO_VERIFY) && r_context()->signing_keyringpath) {
```

Comment:
`CHECK_BUNDLE_NO_VERIFY` normally targets the input bundle only. Maybe we should remove it from the condition here and make the verification of the output bundle depending only on whether `--signing-keyring` was given or not? (Would additionally require to filter out and set the `CHECK_BUNDLE_NO_VERIFY` manually for this case)

It's probably also useful to describe in a short inline comment why this 'hack' is necessary and move this closer to `check_bundle()` to make the context clearer?

Reply:
I've changed it so that it does verification based on whether a keyring was provided.